src/main/java/hudson/remoting/Launcher.java

```diff
@@ -516,19 +516,34 @@ public static void main(InputStream is, OutputStream os, Mode mode, boolean perf
         if (os instanceof StandardOutputStream)
             cb.withProperty(StandardOutputStream.class,os);
 
-        Channel channel = cb.build(is, os);
+        final Channel channel = cb.build(is, os);
         System.err.println("channel started");
         long timeout = 1000 * Long.parseLong(
                 System.getProperty("hudson.remoting.Launcher.pingTimeoutSec", "240")),
              interval = 1000 * Long.parseLong(
                 System.getProperty("hudson.remoting.Launcher.pingIntervalSec", "600"));
+
+        boolean pingFailureAnalyzer = Boolean.parseBoolean(System.getProperty("hudson.remoting.Launcher.pingFailureAnalyzer", "false"));
+
         if (performPing && timeout > 0 && interval > 0) {
-            new PingThread(channel, timeout, interval) {
+            new PingThread(channel, timeout, interval, pingFailureAnalyzer) {
+                @Deprecated
                 @Override
                 protected void onDead() {
                     System.err.println("Ping failed. Terminating");
                     System.exit(-1);
                 }
+
+                @Override
+                protected void onDead(Throwable diagnosis) {
+                    if (this.isPingFailureAnalyzer()) {
```

Comment:
if you shutdown the launcher (with a ctrl-c, service stop etc) - I assume the channel will be closed and `onDead` will be called back.
If this is true then this seems to causing excessive logging.